### PR TITLE
Simulation: Fixes compiliation warnings.

### PR DESCRIPTION
**Pull request by ernieift**
After a compiler update got some warnings and fixed them. No functional
changes.

### CODE REVIEW

File: flight/PiOS.osx/osx/Libraries/FreeRTOS/Source/portable/GCC/Posix/port.c [hunk 1]
```diff
@@ -465,7 +465,7 @@ sigset_t xSignalToBlock;
 	/* Cleanup the mutexes */
 	pthread_mutex_destroy( &xSuspendResumeThreadMutex );
 	pthread_mutex_destroy( &xSwappingThreadMutex );
-	vPortFree( (void *)pxThreads );
+	//vPortFree( (void *)pxThreads );
```
Comment (peabody124) [inline flight/PiOS.osx/osx/Libraries/FreeRTOS/Source/portable/GCC/Posix/port.c]: we don't tend to have commented code. You can just delete this line and the one below.

**Comment by ernieift** [comment]
I know. So i can remove the alloc() function too. It is already commented...

**Comment by peabody124** [comment]
Sure. Go ahead and then I'll merge.

**Comment by mlyle** [comment]
Oh, shoot.  Looks like I have very similar stuff in my pull #1567.

**Comment by peabody124** [comment]
@ernieift how do you feel about closing this in favor of #1567 which has no warnings either?

**Comment by ernieift** [comment]
No problem. I can leave the fixes branch open and close this for now. Have you already tested wether #1567 works? I haven't got the time ;).

**Comment by ernieift** [comment]
Closed because #1567 is the better shot and works.

**Pull request closed without merging**